### PR TITLE
Validate whether URLs are absolute URLs and process accordingly

### PR DESCRIPTION
**Pull request by IsankaSR**
## Purpose

Resolves https://github.com/wso2/product-is/issues/12684 
Validate whether endpoint URLs are absolute URLs and leave the configuration value as it is in such cases.

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java [hunk 1]
```diff
@@ -721,4 +724,14 @@ private void setUserClaimValues(AuthenticatedUser authenticatedUser, String user
 			throw new AuthenticationFailedException(errorMessage, e);
 		}
 	}
+
+	private String buildAbsoluteURL(String redirectUrl) throws URISyntaxException {
```
Comment (ShanChathusanda93) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java:728]: Shall we check the possibility to use the `buildAbsoluteURL` in [1]?

[1] https://github.com/wso2-extensions/identity-outbound-auth-totp/blob/38b34404d0e9834318a3091e3ae2bbf7e7a3c945/component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java#L304

File: component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java [hunk 2]
```diff
@@ -721,4 +726,14 @@ private void setUserClaimValues(AuthenticatedUser authenticatedUser, String user
...
 	}
+
+	private String buildAbsoluteURL(String redirectUrl) throws URISyntaxException, URLBuilderException {
+
+		URI uri = new URI(redirectUrl);
+		if (uri.isAbsolute()) {
+			return redirectUrl;
+		} else {
+			return ServiceURLBuilder.create().addPath(redirectUrl).build().getAbsolutePublicURL();
+		}
```
Comment (AnuradhaSK) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java:737]: else block can be removed as follows.
```suggestion
		if (uri.isAbsolute()) {
			return redirectUrl;
		}
		return ServiceURLBuilder.create().addPath(redirectUrl).build().getAbsolutePublicURL();
```

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticatorTest.java [hunk 3]
```diff
@@ -461,4 +465,52 @@ public IObjectFactory getObjectFactory() {
...
     }
 
-}
\ No newline at end of file
+    private void mockServiceURLBuilder() {
+
+        ServiceURLBuilder builder = new ServiceURLBuilder() {
+
+            String path = "";
+
```
Comment (AnuradhaSK) [inline component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticatorTest.java:473]: remove redundant new lines

File: component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java (hunk 2 above)
Comment (IsankaSR) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java]: Fixed

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticatorTest.java (hunk 3 above)
Comment (IsankaSR) [reply component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticatorTest.java]: Fixed

File: component/authenticator/pom.xml [hunk 4]
```diff
@@ -103,7 +123,21 @@
         <dependency>
             <groupId>org.wso2.carbon.identity.framework</groupId>
             <artifactId>org.wso2.carbon.identity.event</artifactId>
-            <version>${carbon.identity.event.version}</version>
+            <version>${carbon.identity.framework.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.framework</groupId>
+            <artifactId>org.wso2.carbon.identity.core</artifactId>
```
Comment (pamodaaw) [inline component/authenticator/pom.xml:130]: dont we need to have the version here

File: pom.xml [hunk 5]
```diff
@@ -180,6 +185,8 @@
         <carbon.commons.version>4.4.8</carbon.commons.version>
         <carbon.commons.imp.pkg.version>[4.4.0, 5.0.0)</carbon.commons.imp.pkg.version>
         <!--Carbon identity version-->
+        <carbon.identity.framework.version>5.18.187</carbon.identity.framework.version>
+
```
Comment (ShanChathusanda93) [inline pom.xml:189]: Additional new line?

**Review by ShanChathusanda93 — APPROVED** [review]

**Pull request merged**